### DISCUSSION TITLE
project management changes needed

### DISCUSSION THREAD

**Discussion by totaam** [Xpra-org/xpra#3949] [Announcements]
_As originally posted on the mailing list: [project management changes needed](https://lists.devloop.org.uk/pipermail/shifter-users/2023-July/002987.html)_ (2023-07)

---

Xpra is being used extensively for critical deployments, worldwide, by 
individuals as well as some of the most prestigious organizations on the 
planet.

Unfortunately, it seems that very few actually want to contribute to the 
project and this means that the project's funding has not been 
sustainable for a while.
I was hoping that at least some of the more serious enquiries would 
eventually lead somewhere, but it seems they all assumed that someone 
else would.

Unless things improve in the next few months, [I intend to](https://github.com/Xpra-org/xpra/issues/3948):
* archive the [mailing list](https://lists.devloop.org.uk/pipermail/shifter-users/) and switch over to [github discussions](https://github.com/orgs/Xpra-org/discussions),
* move all the package [https://xpra.org/dist/](downloads) to a third party hosting service (TBD),
* redirect [xpra.org](https://xpra.org/) to [github](https://github.com/Xpra-org/xpra),
* finally release [version 5.0](https://github.com/Xpra-org/xpra/tree/v5.x) and a few minor updates after that,
* give commit, release and key privileges to more contributors,
* decommission the hardware currently hosting a lot of these services,
* step down as maintainer

Despite my best efforts to welcome contributions to the project and make 
the codebase more approachable, it seems unlikely that anyone will be 
able to take over in the short term.
It is slightly frightening to think that users will be left with the 
already sub-par downstream packaging for a project of this importance 
and size.
I do feel guilty for this messy state of affairs, so I will gladly 
listen to any suggestions on how best to protect the future of this 
project before I move on.

No matter what happens in the next few weeks / months, I would like to 
take this opportunity to thank all past and present sponsors, and all 
those who have helped the project until now through contributions of any 
kind.

---

* [current and past sponsors](https://github.com/Xpra-org/xpra/blob/master/docs/SPONSORS.md)  
* [sponsor the project](https://github.com/sponsors/totaam)

**Comment by ehfd** [comment]
> move all the package https://xpra.org/dist/ to a third party hosting service (TBD)

https://blog.packagecloud.io/packagecloud-loves-oss/

https://help.cloudsmith.io/docs/open-source-hosting-policy

https://cloudrepo.io/pricing.html

Good candidates for the "third party hosting service"

https://www.rabbitmq.com/install-rpm.html

RabbitMQ is a pretty large project, and they receive support from them.

**Reply by ehfd** [reply]
@totaam 

https://github.com/VirtualGL/virtualgl/issues/240
https://github.com/VirtualGL/virtualgl/issues/225

I would like to say that VirtualGL moved from SourceForge to GitHub (for releases) and PackageCloud (for DEB/RPM)'s free open-source offering. It's not documented on the provider's website so they encourage to email their team.

**Reply by totaam** [reply]
@ehfd I didn't have time to migrate all the services in time this year, so I have paid the hosting fees for another year...
cloudrepo looks like one of the better options, so maybe soon.

**Comment by adamhotep** [comment]
Thank you for all of your hard work. I hope a worthy maintainer will step up.

**Comment by totaam** [comment]
@ehfd thanks, but none of these look like a good fit - I'll read again later.
Our downloads grow to hundreds of gigabytes and include more than just the xpra packages themselves. Also, I am not going to sign up for any kind of "14-day free trial".
The non-free prices are eye-opening though! (over a factor of 10 more than what the current solution costs!)

---

Meanwhile:
* [5.0 has been released](https://github.com/Xpra-org/xpra/releases/tag/v5.0)
* https://xpra.org/ now redirects to https://github.com/Xpra-org/xpra (and the welcome page has been improved somewhat - screenshots added) - but not https://xpra.org/src/ (obviously)
* the mailing list is beeing archived here: https://xpra.org/list/ (search and replace in progress)

So, almost complete.

**Reply by ehfd** [reply]
> thanks, but none of these look like a good fit - I'll read again later.

I believe Xpra qualifies as a relatively large project, so wouldn't hurt to email them to confirm what they can offer for free to open-source projects. RabbitMQ would certainly take much larger bandwidth and storage than Xpra.

Meanwhile, I have been searching for a way to use GitHub as an RPM/DEB repository without much avail.

Technically, it would be possible to hack the URLs inside `repodata` (https://sourceforge.net/projects/turbovnc/files/repodata/) paths to point to GitHub release files. But I found no reliable sources to support that.

**Reply by ehfd** [reply]
https://help.cloudsmith.io/docs/open-source-hosting-policy#sponsorship

I was intending this section for you to see.

**Comment by totaam** [comment]
The mailing list search and replace was a little bit difficult (some links are now dead, ie: subversion links), but now mostly complete.

As for hosting, there is another aspect that I should have considered: we have a bunch of redirects from xpra.org URLs to other places.
A plain-www hosting probably won't cover that.
And we currently have `rsync` read-only access.

**Reply by MayeulC** [reply]
Hi totaam, as a long time (but relatively infrequent) user of Xpra, thank you for everything you did for the project.

I am quite surprised that a plain http server would not cover the needs of the project. I see many open source project getting by just fine with GitHub pages (or GitLab pages, Codeberg pages are a bit less reliable for now). Cloudfare (free tier) can also reduce bandwidth usage and server load, if that is something you wan to consider; it works very well for static files.

Now, about redirect specifically:
* http-equiv: works for most use-cases except "dumb" programs such as curl. I think this is good enough, maintainers can update some upstream URLs when they break, such breakage is obvious. Works for SEO as well as user browsers.
* GitLab Pages [supports](https://docs.gitlab.com/user/project/pages/redirects/) HTTP redirects (up to a instance-defined limit, the doc says 1000 redirects / 64KB by default).
* If doable, all redirects could be hosted on a subdomain hosted elsewhere.

While I am not a web dev specialist, I have quite a bit of experience with these services and can help you out if you need to (and/or basic HTML+CSS :))

**Reply by totaam** [reply]
> I am quite surprised that a plain http server would not cover the needs of the project

It would and if in fact, it does.

> I see many open source project getting by just fine with GitHub pages

I doubt that they're providing repositories with hundreds of gigabytes of packages that aren't arctifacts produced by github workflows.

> Cloudfare (free tier) can also reduce bandwidth usage and server load

It does, somewhat. Not by much.

**Comment by totaam** [comment]
For the record, we have a single [Xpra appreciation thread](https://github.com/orgs/Xpra-org/discussions/3780), but I do get many of these:
* [_Additionally, I have no interest in linking my identity and interests for Microsoft....  I'm not going to buy you a coffee through this Microsoft platform though..._](https://github.com/Xpra-org/xpra/issues/3863#issuecomment-1597134014)
* more dislike of github received by email:
>> the mailing list itself will be replaced by github's discussions:
> 
> _Then I'm out of here._

It's "funny" how people find the time and energy to complain about decisions they never contributed to.

**Reply by ehfd** [reply]
Honestly, developers only want a stable platform where everything is persistently available and won't disappear one day. GitHub is the only option for this because SourceForge has almost failed us.
Centralization of code is a concern but not like we have an option.

**Comment by EgosOwn** [comment]
Sad to see, but understandable decision. thank you for your immense effort.

**Comment by totaam** [comment]
I would like to post this update to the mailing list, but I can't, because of gmail: #3555
I will be updating the [sponsors page](https://github.com/Xpra-org/xpra/blob/master/docs/SPONSORS.md).
It looks like the funding will be sufficient until the end of 2024.
A big thank you to our existing and new sponsors.
